### PR TITLE
[6346] Added tests

### PR DESCRIPTION
**Pull request by KarolJagodzinski**
**Ticket**:[BAU-6346](https://bugzilla.stxnext.pl/show_bug.cgi?id=6346)
Added tests for views and utils

### CODE REVIEW

File: src/presence_analyzer/tests.py [hunk 1]
```diff
@@ -53,6 +52,43 @@ def test_api_users(self):
...
+        resp = self.client.get('/api/v1/mean_time_weekday/10')
+        self.assertEqual(resp.status_code, 200)
+        self.assertEqual(resp.content_type, 'application/json')
+        data = json.loads(resp.data)
+        self.assertEqual(len(data), 7)
+        self.assertListEqual(data[6], [u'Sun', 0])
+
+    def test_api_mean_time2(self):
+        """
+        test api mean time weekday view for unexisting user
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: How comment should look ```Test api mean time weekday view for unexisting user```.

File: src/presence_analyzer/tests.py [hunk 2]
```diff
@@ -53,6 +52,43 @@ def test_api_users(self):
...
+    def test_api_mean_time1(self):
+        """
+        Test api mean time weekday for existing user.
+        """
+        resp = self.client.get('/api/v1/mean_time_weekday/10')
+        self.assertEqual(resp.status_code, 200)
+        self.assertEqual(resp.content_type, 'application/json')
+        data = json.loads(resp.data)
+        self.assertEqual(len(data), 7)
+        self.assertListEqual(data[6], [u'Sun', 0])
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: Check whole list and then you don't need ```self.assertEqual(len(data), 7)```

File: src/presence_analyzer/tests.py [hunk 3]
```diff
@@ -53,6 +52,43 @@ def test_api_users(self):
         self.assertEqual(len(data), 2)
         self.assertDictEqual(data[0], {u'user_id': 10, u'name': u'User 10'})
 
+    def test_api_mean_time1(self):
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: I don't understand why you numbering this methods.

File: src/presence_analyzer/tests.py (hunk 2 above)
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: ```from __future__ import unicode_literals``` and then you don't need ```u```

File: src/presence_analyzer/tests.py [hunk 4]
```diff
@@ -86,6 +122,52 @@ def test_get_data(self):
...
+        self.assertEqual(utils.interval(time, time2), 3720)
+
+    def test_mean_time_with_list(self):
+        """
+        Test calculating arithhmetic mean for list
+        """
+        items = [1, 2, 3]
+        self.assertEqual(utils.mean(items), 2.0)
+
+    def test_mean_time_without_list(self):
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: It is just another case, not another test.

File: src/presence_analyzer/main.py [hunk 5]
```diff
@@ -10,9 +11,8 @@
     os.path.dirname(__file__), '..', '..', 'runtime', 'data', 'sample_data.csv'
 )
 
-
 app = Flask(__name__)  # pylint: disable=invalid-name
 app.config.update(
-    DEBUG = True,
-    DATA_CSV = MAIN_DATA_CSV
+    DEBUG=True,
+    DATA_CSV=MAIN_DATA_CSV
```
Comment (dpolcyn) [inline src/presence_analyzer/main.py]: Changes from pervious ticket?

File: src/presence_analyzer/tests.py [hunk 6]
```diff
@@ -51,7 +52,53 @@ def test_api_users(self):
...
+        self.assertEqual(resp.status_code, 404)
+
+    def test_api_presence_weekday(self):
+        """
+        Test api presence weekday view existing user.
+        """
+        resp = self.client.get('/api/v1/presence_weekday/1')
+        self.assertEqual(resp.status_code, 404)
+
+    def test_api_presence_weekday2(self):
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: You still numbering fuctions. why not ```test_api_presence_weekday_404```?

File: src/presence_analyzer/main.py (hunk 5 above)
Comment (KarolJagodzinski) [inline src/presence_analyzer/main.py]: Yep, lost in git :)

File: src/presence_analyzer/tests.py [hunk 7]
```diff
@@ -51,7 +52,53 @@ def test_api_users(self):
...
+        Test api mean time weekday view for existing user.
+        """
+        resp = self.client.get('/api/v1/mean_time_weekday/10')
+        self.assertEqual(resp.status_code, 200)
+        self.assertEqual(resp.content_type, 'application/json')
+        data = json.loads(resp.data)
+        self.assertEqual(
+            data,
+            [
+                ["Mon", 0],
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: 'Mon'

File: src/presence_analyzer/tests.py [hunk 8]
```diff
@@ -86,6 +133,53 @@ def test_get_data(self):
...
+            utils.group_by_weekday(items),
+            [[7200], [], [10800], [], [], [], []]
+        )
+
+    def test_seconds_since_midnight(self):
+        """
+        Test calculating amount of seconds since midnight
+        """
+        time = datetime.time(1, 2, 3)
+        self.assertEqual(utils.seconds_since_midnight(time), 3723)
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: More cases

File: src/presence_analyzer/tests.py [hunk 9]
```diff
@@ -86,6 +142,86 @@ def test_get_data(self):
             datetime.time(9, 39, 5)
         )
 
+    def test_group_by_weekday(self):
+        """
+        Groups entries by weekday
+        """
+        items = {
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: Why not
```
sample_data = utils.get_data()
result = utils.group_by_weekday(sample_data[10])
```

File: src/presence_analyzer/tests.py [hunk 10]
```diff
@@ -86,6 +142,86 @@ def test_get_data(self):
...
+        self.assertEqual(
+            utils.group_by_weekday(items),
+            [[7200], [], [10800], [], [], [], []]
+        )
+
+    def test_seconds_since_midnight(self):
+        """
+        Test calculating amount of seconds since midnight.
+        """
+        time = datetime.time(1, 2, 3)
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: You don't have to declare one variable for one test. You can just override it after check in assert.

File: src/presence_analyzer/tests.py [hunk 11]
```diff
@@ -86,6 +142,86 @@ def test_get_data(self):
...
+        """
+        Test calculating intervals between two datetime.time obj.
+        """
+        time = datetime.time(0, 0, 3)
+        time2 = datetime.time(1, 2, 3)
+        self.assertEqual(utils.interval(time, time2), 3720)
+        self.assertEqual(utils.interval(time2, time), -3720)
+        self.assertEqual(utils.interval(time2, time2), 0)
+
+    def test_correct_intervals(self):
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: Still wrong, why not ```test_intervals```

File: src/presence_analyzer/tests.py [hunk 12]
```diff
@@ -51,7 +52,62 @@ def test_api_users(self):
...
+        Test api presence weekday view for existing user.
+        """
+        resp = self.client.get('/api/v1/presence_weekday/10')
+        self.assertEqual(resp.status_code, 200)
+        self.assertEqual(resp.content_type, 'application/json')
+        data = json.loads(resp.data)
+        self.assertEqual(
+            data,
+            [
+                ["Weekday", "Presence (s)"],
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: 'Weekday'....

File: src/presence_analyzer/tests.py [hunk 13]
```diff
@@ -86,6 +142,66 @@ def test_get_data(self):
...
+            [[], [30047], [24465], [23705], [], [], []]
+        )
+
+    def test_group_by_weekday2(self):
+        """
+        Groups entries by weekday
+        """
+        sample_data = utils.get_data()
+        result = utils.group_by_weekday(sample_data[10])
+        self.assertEqual(
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: I wrote it before... This is another case and should be in ```test_group_by_weekday```

File: src/presence_analyzer/tests.py [hunk 14]
```diff
@@ -86,6 +142,66 @@ def test_get_data(self):
             datetime.time(9, 39, 5)
         )
 
+    def test_group_by_weekday(self):
+        """
+        Groups entries by weekday
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: Full stop on the end of docstring.

File: src/presence_analyzer/tests.py [hunk 15]
```diff
@@ -86,6 +142,66 @@ def test_get_data(self):
...
+        """
+
+        sample_data = utils.get_data()
+        result = utils.group_by_weekday(sample_data[10])
+        self.assertListEqual(
+            result,
+            [[], [30047], [24465], [23705], [], [], []]
+        )
+
+    def test_group_by_weekday2(self):
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: 2?

File: src/presence_analyzer/tests.py [hunk 16]
```diff
@@ -51,7 +52,62 @@ def test_api_users(self):
...
+            [
+                ['Weekday', 'Presence (s)'],
+                ['Mon', 0],
+                ['Tue', 30047.0],
+                ['Wed', 24465.0],
+                ['Thu', 23705.0],
+                ['Fri', 0],
+                ['Sat', 0],
+                ['Sun', 0],
+            ])
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: ```)``` should be in next line.

**Pull request merged**